lib/ansible/plugins/inventory/vmware.py

```diff
@@ -0,0 +1,238 @@
+        except vim.fault.InvalidLogin as e:
+            raise AnsibleError("Unable to log on to vCenter or ESXi API at %s:%s as %s: %s" % (self.hostname, self.port, self.username, e.msg))
+        except vim.fault.NoPermission as e:
+            raise AnsibleError("User %s does not have required permission"
+                               " to log on to vCenter or ESXi API at %s:%s : %s" % (self.username, self.hostname, self.port, e.msg))
+        except (requests.ConnectionError, ssl.SSLError) as e:
+            raise AnsibleError("Unable to connect to vCenter or ESXi API at %s on TCP/%s: %s" % (self.hostname, self.port, e))
+        except vmodl.fault.InvalidRequest as e:
+            # Request is malformed
+            raise AnsibleError("Failed to get a response from server %s:%s as "
+                               "request is malformed: %s" % (self.hostname, self.port, e.msg))
+        except Exception as e:
+            raise AnsibleError("Unknown error while connecting to vCenter or ESXi API at %s:%s : %s" % (self.hostname, self.port, e))
+
+        if service_instance is None:
+            raise AnsibleError("Unknown error while connecting to vCenter or ESXi API at %s:%s" % (self.hostname, self.port))
+
+        atexit.register(connect.Disconnect, service_instance)
+        return service_instance.RetrieveContent()
+
+    def verify_file(self, path):
+        """
+        Function to verify plugin configuration file and mark this plugin active
+        Args:
+            path: Path of configuration YAML file
+
+        Returns: True if everything is correct, else False
+
+        """
+        valid = False
+        if super(InventoryModule, self).verify_file(path):
+            if path.endswith(('vmware.yaml', 'vmware.yml')):
+                valid = True
+        if not HAS_REQUESTS or not HAS_PYVMOMI:
+            valid = False
+        return valid
+
+    def parse(self, inventory, loader, path, cache=True):
+        """
+        parses the inventory file
+        """
+
+        super(InventoryModule, self).parse(inventory, loader, path, cache=cache)
+
+        # Get Cache key
+        cache_key = self.get_cache_key(path)
+
+        config_data = self._read_config_data(path)
+
+        # set _options from config data
+        self._consume_options(config_data)
+
+        source_data = None
+        if cache:
+            cache = self._options.get('cache')
+
+        self._set_params()
+        self.content = self.login()
+
+        update_cache = False
+        if cache:
+            try:
+                source_data = self.cache.get(cache_key)
+            except KeyError:
+                update_cache = True
+
+        if not cache or update_cache:
+            self._populate_from_source()
+
+    def _populate_from_source(self):
+        """
+        Function to populate inventory data from direct source
+
+        """
+
+        cacheable_results = {'_meta': {'hostvars': {}}}
```

Comment:
unused

Reply:
I am yet to write caching.

Reply:
still `_meta` is a 'script plugin thing' not related to any other inventory plugin